lib/vanagon/platform/deb.rb

```diff
@@ -29,10 +29,15 @@ def generate_package(project) # rubocop:disable Metrics/AbcSize
       # @param workdir [String] working directory to stage the evaluated templates in
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
-      def generate_packaging_artifacts(workdir, name, binding)
+      # @param project [String] Vanagon::Project we are building for
+      def generate_packaging_artifacts(workdir, name, binding, project) # rubocop:disable Metrics/AbcSize
```

Comment:
You'll need to update this in a few more places to make sure all the generate_packaging_artifacts include the project

```
lib/vanagon/platform/osx.rb
67:      def generate_packaging_artifacts(workdir, name, binding) # rubocop:disable Metrics/AbcSize

lib/vanagon/platform/solaris_11.rb
43:      def generate_packaging_artifacts(workdir, name, binding)

lib/vanagon/platform/solaris_10.rb
84:      def generate_packaging_artifacts(workdir, name, binding)

lib/vanagon/platform/windows.rb
39:      def generate_packaging_artifacts(workdir, name, binding)

Reply:
Can we make it an optional parameter?

Reply:
Not super great with Ruby so not sure if that would be possible/a good idea

Reply:
So, ruby has optional parameters but I'm not sure it's going to buy us anything here.

There are two different `generate_packaging_artifacts` right now: one defined in `Vanagon::Project`, which is the method that's initially called when vanagon runs. That method calls the `generate_packaging_artifacts` for the `Vanagon::Platform`. There's no generic `generate_packaging_artifacts`, only the `generate_packaging_artifacts` methods defined for each platform. So, if we don't update all of the methods it won't be able to find it for anything other than rpms and debs

Reply:
ohhh gotcha gotcha